includes/class-sensei-feature-flags.php

```diff
@@ -25,21 +25,58 @@ class Sensei_Feature_Flags {
 	 *
 	 * @var array
 	 */
-	private $feature_flags;
+	private $feature_flags = [];
+
+	/**
+	 * Default feature flags constant.
+	 */
+	private const DEFAULT_FEATURE_FLAGS = [
+		'production'  => [
+			'enrolment_provider_tooltip' => false,
+			'email_customization'        => true,
+			'course_outline_ai'          => false,
+		],
+		'development' => [
+			'enrolment_provider_tooltip' => false,
+			'email_customization'        => true,
+			'course_outline_ai'          => true,
+		],
+	];
 
 	/**
 	 * Sensei_Feature_Flags constructor.
+	 *
+	 * @internal
 	 */
 	public function __construct() {
-		$this->feature_flags = [];
+		add_action( 'init', [ $this, 'register_scripts' ], 9 );
+	}
+
+	/**
+	 * Register scripts.
+	 *
+	 * @internal
+	 *
+	 * @since $$next-version$$
+	 */
+	public function register_scripts() {
+		wp_register_script( 'sensei-feature-flags', '' ); // phpcs:ignore WordPress.WP.EnqueuedResourceParameters -- Intended, this is a placeholder script.
+
+		wp_add_inline_script(
+			'sensei-feature-flags',
+			'window.sensei = window.sensei || {}; ' .
+			'window.sensei.feature_flags = ' . wp_json_encode( $this->get_feature_flags() ) . ';'
```

Comment:
I think is possible to reduce the lines 67-68 to:


We just need to avoid the edge case if we don't have feature flags, because it can generate a error like:

`window.sensei.feature_flags =;` 
which breaks the application.

Maybe we should wrap it in a try-catch?

Reply:
Hm, I wasn't able to break it. I've tested the code with no flags and the result was `window.sensei.feature_flags = [];` which didn't trigger any errors. Could you double-check?